ext/eglgles/gstegladaptation.c

```diff
@@ -584,35 +606,55 @@ gst_egl_adaptation_init_egl_surface (GstEglAdaptationContext * ctx,
       break;
     case GST_VIDEO_FORMAT_BGR:
     case GST_VIDEO_FORMAT_BGRx:
-    case GST_VIDEO_FORMAT_BGRA:
       frag_prog = g_strdup_printf (frag_REORDER_prog, 'b', 'g', 'r');
       free_frag_prog = TRUE;
       ctx->n_textures = 1;
       texnames[0] = "tex";
       break;
+    case GST_VIDEO_FORMAT_BGRA:
+      frag_prog = g_strdup_printf (frag_REORDER_full_prog, 'b', 'g', 'r', 'a');
+      free_frag_prog = TRUE;
+      ctx->n_textures = 1;
+      texnames[0] = "tex";
+      break;
     case GST_VIDEO_FORMAT_xRGB:
-    case GST_VIDEO_FORMAT_ARGB:
       frag_prog = g_strdup_printf (frag_REORDER_prog, 'g', 'b', 'a');
       free_frag_prog = TRUE;
       ctx->n_textures = 1;
       texnames[0] = "tex";
       break;
+    case GST_VIDEO_FORMAT_ARGB:
+      frag_prog = g_strdup_printf (frag_REORDER_full_prog, 'g', 'b', 'a', 'r');
+      free_frag_prog = TRUE;
+      ctx->n_textures = 1;
+      texnames[0] = "tex";
+      break;
     case GST_VIDEO_FORMAT_xBGR:
-    case GST_VIDEO_FORMAT_ABGR:
       frag_prog = g_strdup_printf (frag_REORDER_prog, 'a', 'b', 'g');
       free_frag_prog = TRUE;
       ctx->n_textures = 1;
       texnames[0] = "tex";
       break;
+    case GST_VIDEO_FORMAT_ABGR:
+      frag_prog = g_strdup_printf (frag_REORDER_full_prog, 'a', 'b', 'g', 'r');
```

Comment:
Possibly little memory leak because I don't think that frag_prog will be released later (because in other cases this pointer holds a static const array).

This is a little inconsistency because frag_prog holds both dynamic and static const arrays. IMHO.